common/membership/hashring_test.go

```diff
@@ -128,15 +128,40 @@ func TestRefreshUpdatesRingOnlyWhenRingHasChanged(t *testing.T) {
 
 }
 
+func TestRefreshWillNotifySubscribers(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	pp := NewMockPeerProvider(ctrl)
+	pp.EXPECT().Subscribe(gomock.Any(), gomock.Any()).Times(1)
+	pp.EXPECT().GetMembers("test-service").AnyTimes()
+
+	changed := &ChangedEvent{
+		HostsAdded:   []string{"a"},
+		HostsUpdated: []string{"b"},
+		HostsRemoved: []string{"c"},
+	}
+
+	hr := newHashring("test-service", pp, log.NewNoop(), metrics.NoopScope(0))
+	hr.Start()
+	var changeCh = make(chan *ChangedEvent)
+
+	err := hr.Subscribe("notifyMe", changeCh)
```

Comment:
can you add multiple subscribers just to make sure iteration works

Reply:
yep, it's a good idea. thanks

Reply:
tests are updated